packages/quantic/force-app/main/default/lwc/quanticFoldedResultList/quanticFoldedResultList.js

```diff
@@ -170,7 +170,15 @@ export default class QuanticFoldedResultList extends LightningElement {
   }
 
   get collections() {
-    return this?.state?.results || [];
+    // We need to add a unique key to each result list to make sure to re-render the LWC when the results change.
+    // If the unique key is only the result uniqueId, the LWC will not re-render when the results change AND the same result is still in the results.
+    const searchResponseId = this?.state?.searchResponseId || Math.random();
+    return (
+      this?.state?.results?.map((collection) => ({
+        ...collection,
+        keyResultList: `${searchResponseId}_${collection.result.uniqueId}`,
```

Comment:
For clarity if you prefer, since here we are giving a unique key for each result.
```suggestion
        resultKey: `${searchResponseId}_${collection.result.uniqueId}`,
```